install.sh

```diff
@@ -0,0 +1,279 @@
+#!/bin/bash -eu
+# Software License Agreement (BSD)
+#
+# Author    Tony Baltovski <tbaltovski@clearpathrobotics.com>
+# Copyright (c) 2020, Clearpath Robotics, Inc., All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without modification, are permitted provided that
+# the following conditions are met:
+# * Redistributions of source code must retain the above copyright notice, this list of conditions and the
+#   following disclaimer.
+# * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+#   following disclaimer in the documentation and/or other materials provided with the distribution.
+# * Neither the name of Clearpath Robotics nor the names of its contributors may be used to endorse or
+#   promote products derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED
+# WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+# PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+# TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+#set -x
+
+echo "Starting ROS installion"
+
+# Get Ubuntu version
+ubuntu_version=`lsb_release -sc`
+
+echo ""
+echo -e "\e[39mChecking your Ubuntu version\e[0m"
+echo -e "\e[39mDetected Ubuntu: $ubuntu_version\e[0m"
+
+case $ubuntu_version in
+  "xenial" )
+    ros_version="kinetic"
+  ;;
+  "bionic" )
+    ros_version="melodic"
+  ;;
+  *)
+    echo -e "\e[31mERROR: Unsupported Ubuntu version: $ubuntu_version\e[0m"
+    exit 0
+esac
+
+echo -e "\e[32mUbuntu ${ubuntu_version} is supported, proceeding to install ROS ${ros_version}\e[0m"
+
+echo ""
+echo -e "\e[39mWhich computing platform are you installing on?"
+echo -e "[1]\e[32m Nvidia Jetson Xavier NX\e[0m"
+echo -e "[2]\e[32m Nvidia Jetson Nano\e[0m"
+echo -e "[3]\e[32m Nvidia Jetson AGX Xavier\e[0m"
+echo -e "[4]\e[32m Nvidia Jetson TX2\e[0m"
```

Comment:
Is it worth having an "other" category in case people are using this script to set up a non-Jetson platform (e.g. their own personal laptop)?
Also, is there any nice way of exposing this option as a command-line flag so that it's easier to run the script non-interactively?  If the platform flag is set, don't ask this question. But if it's omitted then run the script interactively

Reply:
Sure but that requires more changes such as not installing envar/udev/config files.
There can be but that would again take additional work.

Reply:
Let's leave it alone for now, and we can revisit this as a future-feature later.